include/correction.h

```diff
@@ -43,8 +45,40 @@ class Formula {
   private:
     std::string expression_;
     ParserType type_;
-    std::vector<int> parameterIdx_;
-    mutable std::unique_ptr<TFormula> evaluator_;
+    std::vector<int> variableIdx_;
+
+    static std::map<ParserType, peg::parser> parsers_;
+    static std::mutex parsers_mutex_; // could be one per parser, but this is good enough
```

Comment:
Have you done large scale threading tests to see how this affect performance?

Reply:
No, but I hope the number of Formulas to compile is << the number of evaluations, so such contention is rare. Unless someone is doing something disgusting like re-loading the data each event...

Reply:
Further, the number of parsers is very small (currently 1)

Reply:
Actually the map is more or less pointless: if I wrapped the peg::parser with a shim class I could construct that as a static const member.